widget/base_renderer.go

```diff
@@ -1,27 +0,0 @@
-package widget
-
-import (
-	"image/color"
-
-	"fyne.io/fyne"
-	"fyne.io/fyne/theme"
-)
-
-type baseRenderer struct {
```

Comment:
I understand why this was moved into internal, however this convenience is no longer unaccessible for developers creating custom widgets. We could expose it as:
```
package widget

import "fyne.io/fyne/internal/widget"

type BaseRenderer struct {
  widget.baseRenderer
}
```
What do you think?

Reply:
I think the advantage is too small to do so.
@andydotxyz convinced me to _not_ add such a construct for the base widget where the advantage would be much bigger. Therefore I would not do that here and now.

Reply:
I think this was not available to any custom widget creation because it was not exposed. The conversation I had with @toaster largely came to the conclusion that someone writing a widget could be expected to write a complete renderer, they do not have the same inheritance properties that widget objects do. Additionally if we make it too "simple" or "static" then there is a much higher chance that third party widgets would not respond well on refresh or theme change (I guess?).

There is another discussion about making it easier to create widgets with trivial renderers at #709 but honestly I am coming to the view that such enablers might encourage things that we don't think are good ideas (such as realising that customisation is difficult resorting to everything becoming a clickable image using a png that does not reflect theme colours)...

It's probably quite a big discussion but given that this was not exposed we can always decide to add your proposed solution in the future.